solr/core/src/java/org/apache/solr/handler/CatStream.java

```diff
@@ -98,7 +98,11 @@ public void setStreamContext(StreamContext context) {
 
     this.chroot = core.getCoreContainer().getUserFilesPath();
     if (! Files.exists(chroot)) {
-      throw new IllegalStateException(chroot + " directory used to load files must exist but could not be found!");
+      try {
```

Comment:
Please no; see JIRA.

Reply:
Hmm. So I reverted to the old code, simply thowing an exception and not trying to create anything. However, that is a back-compat break. Earlier, this exception would never be throws since the folder would always exist. Don't know what CatStream is or whether it makes sense to use it with an empty / no 'userfiles' though.

Reply:
CC @gerlowskija on userfiles in general